pkg/services/hcloud/server/server.go

```diff
@@ -571,38 +579,40 @@ func (s *Service) reconcileLoadBalancerAttachment(ctx context.Context, server *h
 		return nil
 	}
 
-	loadBalancerAddServerTargetOpts := hcloud.LoadBalancerAddServerTargetOpts{
-		Server:       server,
-		UsePrivateIP: &hasPrivateIP,
-	}
-
-	if _, err := s.scope.HCloudClient.AddTargetServerToLoadBalancer(
-		ctx,
-		loadBalancerAddServerTargetOpts,
-		&hcloud.LoadBalancer{
-			ID: s.scope.HetznerCluster.Status.ControlPlaneLoadBalancer.ID,
-		}); err != nil {
-		if hcloud.IsError(err, hcloud.ErrorCodeRateLimitExceeded) {
-			conditions.MarkTrue(s.scope.HCloudMachine, infrav1.RateLimitExceeded)
-			record.Event(s.scope.HCloudMachine,
-				"RateLimitExceeded",
-				"exceeded rate limit with calling hcloud function AddTargetServerToLoadBalancer",
-			)
-		}
-		if hcloud.IsError(err, hcloud.ErrorCodeTargetAlreadyDefined) {
-			return nil
+	// Only if server has IPv4, otherwise Hetzner cannot handle it
+	if server.PublicNet.IPv4.IP != nil {
```

Comment:
Won't this if-statement break the use of a private IPv4 addresses in the Load Balancer as well? This seems because the `hasPrivateIP`-var is not taken into consideration in this if-statement. 

Or, doesn't Hetzner support private IPv4 adresses without public IPv4 being attached?

Reply:
Hi @Wouter0100! This is a great catch, thank you! 
We would have found this problem before the next release because we would have added some e2e tests for this setup without IPv4 addresses. But you saved us definitely some time since we don't have to search for the error anymore. Thanks!